### PR TITLE
PathTracer: add bent normals and diagnostic targets.

### PR DESCRIPTION
**Pull request by prideout**

### CODE REVIEW

File: libs/rays/include/rays/PathTracer.h [hunk 1]
```diff
@@ -30,18 +30,32 @@ namespace filament {
 namespace rays {
 
 /**
- * Signals that a region of the generated image has become available, typically used for progress
- * notification. This can be called from any thread.
+ * RenderTarget denotes a client-owned image that the PathTracer renders into.
+ *
+ * All output planes must have 3 channels except AMBIENT_OCCLUSION, which must have 1.
+ */
+enum RenderTarget {
```
Comment (romainguy) [inline libs/rays/include/rays/PathTracer.h]: I'm not sure RenderTarget is the best name here. I can't think of a better one though. BakeData? BakeTarget?

Comment (prideout) [reply libs/rays/include/rays/PathTracer.h]: These images can be written into for non-baking tasks as well (e.g. when invoking the pathtracer with SimpleCamera).  So I don't think bake should be in the name.  Maybe `OutputPlane`?

Comment (prideout) [reply libs/rays/include/rays/PathTracer.h]: Renamed to OutputPlane.

File: libs/rays/include/rays/PathTracer.h [hunk 2]
```diff
@@ -30,18 +30,32 @@ namespace filament {
...
+ * All output planes must have 3 channels except AMBIENT_OCCLUSION, which must have 1.
+ */
+enum OutputPlane {
+    AMBIENT_OCCLUSION = 0,
+    BENT_NORMALS = 1,
+    MESH_NORMALS = 2,
+    MESH_POSITIONS = 3
+};
+
+static constexpr size_t RENDERTARGET_COUNT = 4;
```
Comment (bejado) [inline libs/rays/include/rays/PathTracer.h]: Rename to `OUTPUTPLANE_COUNT`

File: libs/rays/src/PathTracer.cpp [hunk 3]
```diff
@@ -76,8 +75,8 @@ static float3 randomPerp(float3 n) {
 namespace filament {
 namespace rays {
 
-PathTracer::Builder& PathTracer::Builder::renderTarget(image::LinearImage target) {
-    mConfig.renderTarget = target;
+PathTracer::Builder& PathTracer::Builder::outputPlane(OutputPlane target, LinearImage image) {
```
Comment (bejado) [inline libs/rays/src/PathTracer.cpp]: Might be good to note which `OutputPlane`s are required and which are optional? Looks like `AMBIENT_OCCLUSION` is the only required one for now.

File: libs/rays/include/rays/PathTracer.h (hunk 2 above)
Comment (prideout) [reply libs/rays/include/rays/PathTracer.h]: Done

File: libs/rays/src/PathTracer.cpp (hunk 3 above)
Comment (prideout) [reply libs/rays/src/PathTracer.cpp]: Done

**Pull request merged**